dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/internal/CoreFeature.kt

```diff
@@ -560,10 +566,12 @@ internal class CoreFeature(
     }
 
     private fun setupExecutors() {
-        @Suppress("UnsafeThirdPartyFunctionCall") // pool size can't be <= 0
-        uploadExecutorService = LoggingScheduledThreadPoolExecutor(CORE_DEFAULT_POOL_SIZE, internalLogger)
-        @Suppress("UnsafeThirdPartyFunctionCall") // workQueue can't be null
-        persistenceExecutorService = persistenceExecutorServiceFactory(internalLogger)
+        uploadExecutorService = LoggingScheduledThreadPoolExecutor(
+            CORE_DEFAULT_POOL_SIZE,
+            internalLogger,
+            backpressureStrategy
+        )
+        persistenceExecutorService = executorServiceFactory.create(internalLogger, backpressureStrategy)
```

Comment:
It seems here it is quite severe change: currently `persistenceExecutorService` is using an executor where max number of threads is `Runtime.getRuntime().availableProcessors()`. But with this change it will use `BackPressureExecutorService` which is providing a single threaded executor. Is this intended change?

Reply:
I actually wonder why we used an executor with more than 1 thread there ? The way I know is that the producer should be serial and the consumer (uploader) can use more threads. There is no reason to use more threads to write as the file will be locked anyway.

Reply:
It is indeed on purpose. The persistence executor would not gain much by being multi threaded since we have locks to avoid writing the same file concurrently. Since most of the issues happen by spamming the same track, we would essentially have the same bottleneck, with the additional context switching cost of juggling more threads.